src/engine/parser/error.rs

```diff
@@ -0,0 +1,190 @@
+// This file is part of rust-web/twig
+//
+// For the copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+//! Typisation of parser errors.
+
+use std::fmt::{self, Display};
+use error::Error;
+use error::{GeneralizeTo, ErrorCode};
+
+use engine::parser::token::TokenErrorCode;
+use engine::parser::job::{self, cursor};
+use engine::parser::token;
+
+pub type ParserError = Error<ParserErrorCode>;
+pub type NodeError = Error<NodeErrorCode>; // todo move somewhere else??
+
+impl GeneralizeTo<ParserErrorCode> for TokenErrorCode {
+    fn generalize(&self) -> ParserErrorCode { ParserErrorCode::TokenError }
+}
+
+#[allow(dead_code)]
+#[derive(Debug)]
+pub enum ParserErrorCode {
+    Unreachable {
+        reason: String,
+        job: job::JobDump,
+    },
+    MissingExtensions,
```

Comment:
In my codebase:
```
impl Parser {
    pub fn new(twig: &Engine) -> Result<Parser, ParserError> {
        let ext = match twig.extensions() {
            Err(e) => return Err(ParserErrorCode::MissingExtensions
                .at(loc!())
                .caused_by(e)),
            Ok(ext) => ext
        };
 \\...
```

I guess you suggest to pass extensions to the Parser-Constructor directly? Then Engine would throw this error instead of Parser - the error code would move to Engine.

But again, a detail not relevant for the skeleton - I will remove it from the skeleton.